### PR TITLE
Make timeout optional property

### LINKED ISSUE #17: Error when running on production env
Hello,

running the example `https://github.com/zeit/next.js/tree/master/examples/with-next-page-transitions` on production env triggers an error:

> TypeError: Cannot read property 'isRequired' of null
    at Object.<anonymous> (/home/freddy/Workspace/workspace-projects/workspace-together/with-next-page-transitions/node_modules/next-page-transitions/lib/PageTransition.js:300:37)

Which refers to this:  line 300 - PageTransition.js
`timeout: _PropTypes.timeoutsShape.isRequired,`

When looking from the origin of the problem, I found this line on the react-transition-group dependency

`var timeoutsShape = process.env.NODE_ENV !== 'production' ? _propTypes.default.oneOfType([_propTypes.default.number, _propTypes.default.shape({
  enter: _propTypes.default.number,
  exit: _propTypes.default.number
}).isRequired]) : null;`

Note: this problems does not appear on the 1.0.0-alpha.1 release because back then we had on PageTransition.js:

`timeout: _PropTypes.timeoutsShape`

Regards,

Freddy

### PR DESCRIPTION
**Pull request by teleginzhenya**
Resolves #17 and https://github.com/zeit/next.js/issues/6086

### CODE REVIEW

File: src/PageTransition.js [hunk 1]
```diff
@@ -268,9 +268,10 @@ PageTransition.propTypes = {
 
 PageTransition.defaultProps = {
   loadingComponent: null,
-  loadingCallbackName: 'pageTransitionReadyToEnter',
+  loadingCallbackName: 'pagesTransitionReadyToEnter',
```
Comment (nwalters512) [inline src/PageTransition.js]: Why was this prop name changed?

Comment (teleginzhenya) [reply src/PageTransition.js]: Sorry, this is typo, will fix it soon

**Comment by euharrison** [comment]
@nwalters512 This is the same solution for react-transition-group, I agree with you.

ref: https://github.com/reactjs/react-transition-group/blob/c3b7708a536a7625770c5131137ba8ec3aea82a2/src/Transition.js#L446-L450

File: src/PageTransition.js [hunk 2]
```diff
@@ -262,6 +261,11 @@ PageTransition.propTypes = {
     if (props.loadingTimeout) pt = pt.isRequired
     return pt(props, ...args)
   },
+  timeout: (props, ...args) => {
+    let pt = timeoutsShape
+    if (pt) pt = pt.isRequired
+    return pt(props, ...args)
```
Comment (euharrison) [inline src/PageTransition.js]: pt could be `undefined`
```suggestion
    if (pt) {
      pt = pt.isRequired
      return pt(props, ...args)
    }
```

Comment (nwalters512) [reply src/PageTransition.js]: Yep, agreed that this is the right thing to do.

**Comment by nwalters512** [comment]
Closing in favor of #21 - thanks for your work on this though!

**Pull request closed without merging**